Makefile

```diff
@@ -143,16 +143,17 @@ TF_CFLAGS_aarch64	=	-mgeneral-regs-only -mstrict-align
 ASFLAGS_aarch32		=	-march=armv8-a
 TF_CFLAGS_aarch32	=	-march=armv8-a
 
-ASFLAGS			+= 	-nostdinc -ffreestanding -Wa,--fatal-warnings	\
-				-Werror -Wmissing-include-dirs			\
-				-D__ASSEMBLY__ $(ASFLAGS_$(ARCH))		\
-				${DEFINES} ${INCLUDES}
-TF_CFLAGS		+= 	-nostdinc -ffreestanding -Wall			\
-				-Werror -Wmissing-include-dirs			\
-				-std=c99 -c -Os					\
-				$(TF_CFLAGS_$(ARCH))				\
-				${DEFINES} ${INCLUDES}
-TF_CFLAGS		+=	-ffunction-sections -fdata-sections
+# common to CPP, CC, AS
+COMMON_FLAGS		=	${DEFINES} ${INCLUDES} -nostdinc		\
+				-Wmissing-include-dirs -Werror
+
+CPPFLAGS		+=	$(COMMON_FLAGS) -D__ASSEMBLY__
```

Comment:
OK. FYI, I'm guessing your pull request won't be merged this week, as the project's maintainers are out of the office at the moment and will be back next week.